web/src/server_events_dispatch.js

```diff
@@ -294,6 +297,28 @@ export function dispatch_normal_event(event) {
                                     gear_menu.rerender();
                                 }
 
+                                if (
+                                    key === "direct_message_initiator_group" ||
+                                    key === "direct_message_permission_group"
+                                ) {
+                                    if (
+                                        realm.realm_direct_message_permission_group ===
+                                        user_groups.get_user_group_from_name("role:nobody").id
+                                    ) {
+                                        $("#realm_direct_message_initiator_group_widget").prop(
+                                            "disabled",
+                                            true,
+                                        );
+                                    } else {
+                                        $("#realm_direct_message_initiator_group_widget").prop(
+                                            "disabled",
+                                            false,
+                                        );
+                                    }
```

Comment:
Do we disable the `Who can initiate direct message dropdown` after just updating the other setting to `Nobody` before clicking `Save changes` button? If not then I think we should do that and then we can probably extract this code in a function which would disable the value based on the input element value. That function can probably be used in `admin.js` as well.

I guess we have similar functions in `settings_org.js`, so that could be used as a reference.

Reply:
No, it was not disabled until the `Save changes` button was clicked. I have added a function `setting_dropdown_item_click_callback` in `item_click_callback` of the `dropdown_widget` to disable it based on the clicked value.